includes/woopay/class-woopay-session.php

```diff
@@ -338,6 +332,15 @@ private static function get_init_session_request() {
 			$customer_id   = WC_Payments::get_customer_service()->create_customer_for_user( $user, $customer_data );
 		}
 
+		if ( 0 !== $user->ID ) {
+			$currency      = get_user_meta( $user->ID, MultiCurrency::CURRENCY_META_KEY, true );
+			$currency_code = strtoupper( $currency );
+
+			if ( ! empty( $currency_code ) && WC()->session ) {
+				WC()->session->set( MultiCurrency::CURRENCY_SESSION_KEY, $currency_code );
+			}
+		}
```

Comment:
Can we add a comment here explaining why this is necessary? I could see this being an area where someone updating this code in the future has no idea why this is here 🤔

Reply:
Great callout! Added a comment now in latest commit.

Reply:
Looks good!